packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -346,9 +346,8 @@ class _DropdownRoute<T> extends PopupRoute<_DropdownRouteResult<T>> {
     }
 
     if (scrollController == null) {
-      double scrollOffset = 0.0;
-      if (preferredMenuHeight > maxMenuHeight)
-        scrollOffset = selectedItemOffset - (buttonTop - menuTop);
+      final double scrollOffset = (preferredMenuHeight > maxMenuHeight) ?
+        math.max(0.0, selectedItemOffset - (buttonTop - menuTop)) : 0.0;
```

Comment:
what i find weird about this approach is that it's asymmetrical. Why do we clamp at zero but not at some equivalent large offset?

Reply:
I never saw a situation where `selectedItemOffset - (buttonTop - menuTop)` produced values that were incorrectly large, so I didn't see a need for it.

In my testing it looked like this did correctly calculate an offset for all items after the first one. It perfectly aligned them with the dropdown button itself if there was room to scroll to them. But it was consistently setting a negative value for when either nothing or the first item in a list was selected. In short lists that doesn't matter since there's no room to scroll, but as soon as there was a scrollbar then it was wrapping around to the bottom of the list to match the negative `scrollOffset`.

@jslavitz's fix in flutter/flutter#22594 looks more robust to me. If the problem is with the variables in this calculation, updating that makes more sense than clamping off known incorrect bounds.